crawl/pom.py

```diff
@@ -537,20 +538,39 @@ def __init__(self, workspace, artifact_def, dependency, pom_template):
         super(DynamicPomGen, self).__init__(workspace, artifact_def, dependency)
         self.pom_content = workspace.pom_content
         self.pom_template = pom_template
+        self._workspace = workspace
 
     def gen(self, pomcontenttype):
         content = self.pom_template.replace("#{group_id}", self._artifact_def.group_id)
         content = content.replace("#{artifact_id}", self._artifact_def.artifact_id)
         version = self._artifact_def_version(pomcontenttype)
         content = content.replace("#{version}", version)
         content = self._handle_description(content, self.pom_content.description)
+
+        # Update the dependencies according to the overridden file
+        self.dependencies = self._update_overridden_deps(self.dependencies)
```

Comment:
Could the update_overriden deps method be called right after parsing the pinned files, in the workspace module?

Reply:
No, the method needs the output of the pinned deps to override the deps.